code/game/objects/effects/mines.dm

```diff
@@ -37,18 +142,64 @@
 	var/range_heavy = 1
 	var/range_light = 2
 	var/range_flash = 3
+	disarm_product = /obj/item/deployablemine/explosive
+
+/obj/effect/mine/explosive/traitor
+	name = "rubber ducky"
+	desc = "Rubber ducky you're so fine, you make bathtime lots of fuuun. Rubber ducky I'm awfully fooooond of yooooouuuu~"
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "rubberducky"
+	var/sound = 'sound/items/bikehorn.ogg'
+	range_heavy = 2
+	range_light = 3
+	range_flash = 4
+	disarm_time = 250
+	disarm_product = /obj/item/deployablemine/traitor
+
+/obj/effect/mine/explosive/traitor/bigboom
+	range_devastation = 2
+	range_heavy = 4
+	range_light = 8
+	range_flash = 6
+	disarm_product = /obj/item/deployablemine/traitor/bigboom
 
 /obj/effect/mine/explosive/mineEffect(mob/victim)
 	explosion(loc, range_devastation, range_heavy, range_light, range_flash)
 
+/obj/effect/mine/explosive/traitor/mineEffect(mob/victim)
+	playsound(loc, sound, 100, 1)
+	explosion(loc, range_devastation, range_heavy, range_light, range_flash)
 
 /obj/effect/mine/stun
 	name = "stun mine"
 	var/stun_time = 80
+	var/damage = 0
+	disarm_product = /obj/item/deployablemine/stun
+
+/obj/effect/mine/stun/smart
+	name = "smart stun mine"
+	desc = "An advanced mine with IFF features, capable of ignoring people with mindshield implants."
+	smartmine = 1
+	disarm_time = 150
+	disarm_product = /obj/item/deployablemine/smartstun
+
+/obj/effect/mine/stun/smart/adv
+	name = "rapid smart mine"
+	disarm_time = 80
+	disarm_product = /obj/item/deployablemine/rapid
+
+/obj/effect/mine/stun/smart/heavy
+	name = "sledgehammer smart mine"
+	disarm_time = 170
+	stun_time = 230
+	damage = 40
+	disarm_product = /obj/item/deployablemine/heavy
+
 
 /obj/effect/mine/stun/mineEffect(mob/living/victim)
 	if(isliving(victim))
-		victim.Paralyze(stun_time)
+		victim.adjustStaminaLoss(stun_time)
```

Comment:
If you are going to change it from a paralyze to a stam loss, rename the variable to reflect the fact its not a stun length but stamina damage